dashboard/app/helpers/aichat_comprehend_helper.rb

```diff
@@ -8,19 +11,68 @@ def self.create_comprehend_client
 
   # Moderate given text for inappropriate/toxic content using AWS Comprehend client.
   def self.get_toxicity(text, locale)
-    comprehend_response = create_comprehend_client.detect_toxic_content(
-      {
-        text_segments: [{text: text}],
-        language_code: locale,
-      }
-    )
-    categories = comprehend_response.result_list[0].labels
+    text_segment_lists = get_text_segment_lists(text)
+    all_results = []
+    text_segment_lists.each do |list|
+      comprehend_response = create_comprehend_client.detect_toxic_content(
+        {
+          text_segments: list.map {|segment| {text: segment}},
+          language_code: locale,
+        }
+      )
+      all_results.concat(comprehend_response.result_list)
+    end
+
     {
       text: text,
-      toxicity: comprehend_response.result_list[0].toxicity,
-      max_category: categories.max_by(&:score),
+      toxicity: all_results.max_by(&:toxicity).toxicity,
+      max_category: all_results.map(&:labels).flatten.max_by(&:score)
```

Comment:
Is max the fold operation we want long-term?

Reply:
I think so? This tells us the category that had the highest score. Is there another operation we should consider though?